src/manage.c

```diff
@@ -5778,30 +5828,56 @@ get_system_report_types (const char *required_type, gchar ***start,
   gint exit_status;
 
   if (slave_id && strcmp (slave_id, "0"))
-    return get_slave_system_report_types (required_type, start, types,
-                                          slave_id);
+    {
+      scanner_t slave;
+      scanner_type_t slave_type;
 
-  g_debug ("   command: " COMMAND);
+      slave = 0;
+      slave_type = SCANNER_TYPE_NONE;
 
-  if ((g_spawn_command_line_sync (COMMAND,
-                                  &astdout,
-                                  &astderr,
-                                  &exit_status,
-                                  &err)
-       == FALSE)
-      || (WIFEXITED (exit_status) == 0)
-      || WEXITSTATUS (exit_status))
+      if (find_scanner_with_permission (slave_id, &slave, "get_scanners"))
+        return -1;
+      if (slave == 0)
+        return 2;
+
+      if (slave_type == SCANNER_TYPE_GMP)
+        return get_slave_system_report_types (required_type, start, types,
+                                              slave);
+      else
+        {
+          // Assume OSP scanner
```

Comment:
Would like the declaration section before the comments, like we usually do.

Reply:
I've moved the comment in 2486eea576faad720d947181c2b476d357ca14be.

Reply:
thank you.